PhotonSieveDevelopment.js

```diff
@@ -24,92 +24,96 @@ class Sieve {
         this.totalPinholes = totalPinholes
         this.totalArea = totalArea
         this.fNumber = fNumber
+        this.rings = []
+    }
+
+    addRing(ring) {
+        this.rings.push(ring);
+        this.totalArea = this.totalArea + ring.openArea;
+        this.totalPinholes = this.totalPinholes + ring.pinholes.length;
+        let sieveEquivalentAperture = Math.sqrt(this.totalArea / Math.PI) * 2;
+        this.fNumber = FD / sieveEquivalentAperture;

```

Comment:
Moved logic for adding a ring into a function owned by sieve to cleanup main code a bit and show object oriented principles